### PR TITLE
ci: [CNI] Adding aks cluster creation steps for k8s e2e test

### PR DESCRIPTION
**Pull request by vipul-21**
This PR adds creation of aks cluster and replacing the CNI binaries. This is a step to migrate away from aks-engine(deprecated). @jpayne3506  will be adding the datapath to these created clusters.
 
**Reason for Change**:

AKS Engine is deprecated.

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/singletenancy/aks/e2e-job-template.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          - template: e2e-step-template.yaml
+            parameters:
+              name: ${{ parameters.name }}
+              clusterType: ${{ parameters.clusterType }}
+              clusterName: ${{ parameters.clusterName }}
+              arch: ${{ parameters.arch }}
+              os: ${{ parameters.os }}
+              k8sVersion: ${{ parameters.k8sVersion }}
+              version: $(dropgzVersion)
+              windowsOsSku: ${{ parameters.windowsOsSku}}
```
Comment (jpayne3506) [inline .pipelines/singletenancy/aks/e2e-job-template.yaml:41]: Nit: Spacing

File: .pipelines/pipeline.yaml [hunk 2]
```diff
@@ -330,25 +336,45 @@ stages:
...
-      clusterDefinition: "cniLinux1804.json"
-      clusterDefinitionCniTypeKey: "azureCNIURLLinux"
-      clusterDefinitionCniBuildOS: "linux"
-      clusterDefinitionCniBuildExt: ".tgz"
+      displayName: AKS Ubuntu 18.04
+      arch: amd64
+      os: linux
+      clusterType: linux-cniv1-up
+      clusterName: "ubuntu18e2e"
+      k8sVersion: 1.24.9
```
Comment (jpayne3506) [inline .pipelines/pipeline.yaml:347]: Does it not default to 1.24.x? I do not think that we should specify versions as they will get outdated. 

Also 1.24.x is End of Life this month. Should we consider shifting this to ubuntu 22 and 1.25.x+?

File: .pipelines/pipeline.yaml [hunk 3]
```diff
@@ -330,25 +336,45 @@ stages:
...
-      clusterDefinition: "cniWindows1903.json"
-      clusterDefinitionCniTypeKey: "azureCNIURLWindows"
-      clusterDefinitionCniBuildOS: "windows"
-      clusterDefinitionCniBuildExt: ".zip"
+      displayName: AKS Windows 1903
+      arch: amd64
+      os: linux
+      clusterType: windows-cniv1-up
+      clusterName: "win19e2e"
+      windowsOsSku: Windows2019
```
Comment (jpayne3506) [inline .pipelines/pipeline.yaml:357]: Nit: Is this testing windows 1903 or 2019

File: .pipelines/pipeline.yaml (hunk 2 above)
Comment (vipul-21) [reply .pipelines/pipeline.yaml:347]: No by default it is greater than 1.25, the reason for this is aks does not support ubuntu 18 with k8s 1.25.
The pipeline currently is testing ubuntu 18 so we need to test for ubuntu 18.
https://learn.microsoft.com/en-us/azure/aks/cluster-configuration

Regarding the ubuntu 22, you are right we need to support that. Once this is merged. It would be just a matter of adding another set of parameters for Ubuntu 22.

File: .pipelines/pipeline.yaml (hunk 3 above)
Comment (vipul-21) [reply .pipelines/pipeline.yaml:357]: Windows 2019 update version was 1903.

File: .pipelines/singletenancy/aks/e2e-step-template.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        export DROP_GZ_URL=$( make cni-dropgz-image-name-and-tag OS='linux' ARCH=${{ parameters.arch }} CNI_DROPGZ_VERSION=${{ parameters.version }})
+        envsubst < ./test/integration/manifests/cni/cni-installer-v1.yaml | kubectl apply -f -
+        kubectl rollout status daemonset/azure-cni -n kube-system
+        export CNI_IMAGE=$(make cni-plugin-image-name-and-tag OS=${{ parameters.os }} ARCH=${{ parameters.arch }} os_version=${{ parameters.os_version }} CNI_VERSION=${{ parameters.cniVersion }})
+        envsubst < ./hack/manifests/windows-update.yaml | kubectl apply -f -
+        kubectl rollout status daemonset/azure-cni-windows -n kube-system
+      else
+        export DROP_GZ_URL=$( make cni-dropgz-image-name-and-tag OS=${{ parameters.os }} ARCH=${{ parameters.arch }} CNI_DROPGZ_VERSION=${{ parameters.version }})
+        envsubst < ./test/integration/manifests/cni/cni-installer-v1.yaml | kubectl apply -f -
+        kubectl rollout status daemonset/azure-cni -n kube-system
```
Comment (tamilmani1989) [inline .pipelines/singletenancy/aks/e2e-step-template.yaml:52]: have to restart nodes after installing cni.. will that be a separate PR?

File: .pipelines/pipeline.yaml [hunk 5]
```diff
@@ -330,6 +336,27 @@ stages:
...
+    parameters:
+      name: "aks_ubuntu_18_04_linux_e2e"
+      displayName: AKS Ubuntu 18.04
+      arch: amd64
+      os: linux
+      clusterType: linux-cniv1-up
+      clusterName: "ubuntu18e2e"
+      k8sVersion: 1.24.9
+
+  - template: singletenancy/aks/e2e-job-template.yaml
```
[the remark this answers is not in the document]
Comment (vipul-21) [reply .pipelines/pipeline.yaml:349]: Sure let me add both windows 22 and ubuntu 22

File: .pipelines/singletenancy/aks/e2e-step-template.yaml (hunk 4 above)
Comment (vipul-21) [reply .pipelines/singletenancy/aks/e2e-step-template.yaml:52]: Yes, but i think I am still testing the windows part of it so will add to this PR.

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: .pipelines/pipeline.yaml [hunk 6]
```diff
@@ -330,6 +341,48 @@ stages:
...
+    parameters:
+      name: "aks_ubuntu_22_linux_e2e"
+      displayName: AKS Ubuntu 22
+      arch: 'amd64'
+      os: 'linux'
+      clusterType: linux-cniv1-up
+      clusterName: 'ubuntu22e2e'
+      k8sVersion: 1.25
```
[the remark this answers is not in the document]
Comment (vipul-21) [reply .pipelines/pipeline.yaml:362]: I tried latest but it showed error. I added 1.25 as it picked the latest 1.25 supported by aks i.e 1.25.6.
We will remove this once we no longer support ubuntu 18 OR need to update it to 1.26 once we have that as default.

File: test/integration/manifests/cni/cni-installer-v1.yaml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            - deploy
+            - azure-vnet
+            - -o
+            - /opt/cni/bin/azure-vnet
+            - azure-vnet-ipam
+            - -o
+            - /opt/cni/bin/azure-vnet-ipam
+            - azure-vnet-telemetry
+            - -o 
+            - /opt/cni/bin/azure-vnet-telemetry
```
[the remark this answers is not in the document]
Comment (vipul-21) [reply test/integration/manifests/cni/cni-installer-v1.yaml:55]: azure-vnet-telemetry config you mean ?

File: dropgz/build/cniTest.Dockerfile [hunk 8]
```diff
@@ -18,10 +18,13 @@ WORKDIR /dropgz
 COPY dropgz .
 COPY --from=azure-ipam /azure-ipam/*.conflist pkg/embed/fs
 COPY --from=azure-ipam /azure-ipam/bin/* pkg/embed/fs
+COPY --from=azure-vnet /azure-container-networking/cni/azure-$OS.conflist pkg/embed/fs/azure.conflist
 COPY --from=azure-vnet /azure-container-networking/cni/azure-$OS-swift.conflist pkg/embed/fs/azure-swift.conflist
 COPY --from=azure-vnet /azure-container-networking/cni/azure-$OS-swift-overlay.conflist pkg/embed/fs/azure-swift-overlay.conflist
 COPY --from=azure-vnet /azure-container-networking/cni/azure-$OS-swift-overlay-dualstack.conflist pkg/embed/fs/azure-swift-overlay-dualstack.conflist
 COPY --from=azure-vnet /azure-container-networking/bin/* pkg/embed/fs
+COPY --from=azure-vnet /azure-container-networking/telemetry/azure-vnet-telemetry.config pkg/embed/fs
```
Comment (vipul-21) [inline dropgz/build/cniTest.Dockerfile:26]: @rbtr I added the azure conflist and telemetry config for testing cniv1 on the aks clusters. This is for cniTest Dockerfile and I guess we need to change the same in linux Dockerfile(will keep that separate as that has production impact for dropgz release)
cc: @tamilmani1989

**Review by tamilmani1989 — APPROVED** [review]
lgtm

**Review by tamilmani1989 — APPROVED** [review]
